tracer/src/Datadog.Trace/Iast/StackWalker.cs

```diff
@@ -41,9 +47,10 @@ internal static class StackWalker
 
     private static readonly ConcurrentDictionary<string, bool> ExcludedAssemblyCache = new ConcurrentDictionary<string, bool>();
 
-    public static StackFrameInfo GetFrame(StackTrace? externalStack = null)
+    public static StackFrame? GetFrame(ref StackTrace? externalStack)
     {
         var stackTrace = externalStack ?? new StackTrace(DefaultSkipFrames, true);
+        externalStack = stackTrace;
```

Comment:
This seems like confusing behaviour to me 🤔 I think it probably makes more sense to push responsibility for calling `new StackTrace(DefaultSkipFrames, true)` higher up, and simply make the argument non-null instead...

```suggestion
    public static StackFrame? GetFrame(StackTrace stackTrace)
    {
```

You could create a small helper the caller can use if needs be

```csharp
    public static StackTrace CreateDefaultStackTrace() => new(DefaultSkipFrames, true);
```

Reply:
Fixed